micro_dl/input/dataset.py

```diff
@@ -69,21 +93,35 @@ def _augment_image(self, input_image, aug_idx):
          5 - rot 270
         :return np.array image after transformation is applied
         """
+        # We need to flip over different dimensions depending on data format
+        add_dim = 0
+        if self.data_format == 'channels_first':
+            add_dim = 1
 
-        assert len(input_image.shape) == 2, 'current implementation works' \
-                                            'for 2D images only'
         if aug_idx == 0:
             return input_image
         elif aug_idx == 1:
-            trans_image = np.fliplr(input_image)
+            trans_image = np.flip(input_image, 1 + add_dim)
```

Comment:
# flip about axis=1 (which is row in numpy, hence about y)

Reply:
Could we please add these notes? (sorry, i usually get confused with rotation axes, hence trying to add these)

Reply:
Sure!